WalletWasabi/Backend/Models/FilterModel.cs

```diff
@@ -33,21 +33,28 @@ public FilterModel(SmartHeader header, Lazy<GolombRiceFilter> filter)
 
 	public static FilterModel FromLine(string line)
 	{
-		string[] parts = line.Split(':');
-
-		if (parts.Length < 5)
+		try
 		{
-			throw new ArgumentException(line, nameof(line));
-		}
+			string[] parts = line.Split(':');
+
+			if (parts.Length < 5)
+			{
+				throw new ArgumentException(line, nameof(line));
+			}
 
-		uint blockHeight = uint.Parse(parts[0]);
-		uint256 blockHash = uint256.Parse(parts[1]);
-		byte[] filterData = Encoders.Hex.DecodeData(parts[2]);
-		Lazy<GolombRiceFilter> filter = new(() => new GolombRiceFilter(filterData, 20, 1 << 20), LazyThreadSafetyMode.ExecutionAndPublication);
-		uint256 prevBlockHash = uint256.Parse(parts[3]);
-		long blockTime = long.Parse(parts[4]);
+			uint blockHeight = uint.Parse(parts[0]);
+			uint256 blockHash = uint256.Parse(parts[1]);
+			byte[] filterData = Encoders.Hex.DecodeData(parts[2]);
+			Lazy<GolombRiceFilter> filter = new(() => new GolombRiceFilter(filterData, 20, 1 << 20), LazyThreadSafetyMode.ExecutionAndPublication);
+			uint256 prevBlockHash = uint256.Parse(parts[3]);
+			long blockTime = long.Parse(parts[4]);
 
-		return new FilterModel(new SmartHeader(blockHash, prevBlockHash, blockHeight, blockTime), filter);
+			return new FilterModel(new SmartHeader(blockHash, prevBlockHash, blockHeight, blockTime), filter);
+		}
+		catch (FormatException ex)
+		{
+			throw new FormatException("There was an error parsing Filters. Resynchronize needed, please restart Wasabi.", ex);
```

Comment:
Why ask the user to restart Wasabi if this will be shown in the crash reporter after Wasabi was closed already?

```suggestion
			throw new FormatException("An error occurred while parsing the block filters.", ex);
```